Clockwise/VirtualClock.cs

```diff
@@ -53,19 +52,19 @@ private VirtualClock(DateTimeOffset? now = null, [CallerMemberName] string creat
 
             using (var operation = AndConfirmAdvancement(now, time))
             {
-                while (schedule.Count > 0)
+                int length;
```

Comment:
cache the length value so that we don't need to constantly recompute 

(sometimes the JIT is not great about realizing that properties like `.Count` haven't changed)

Reply:
Is this code change intended to compensate for a possible JIT bug?

Reply:
It's intended to compensate for the fact that the JIT is not good at analyzing functions for side-effecting operations.

The "Count" getter *could possibly* mutate the list, and if the JIT doesn't want to look inside the getter to see if it side-effects, then it makes the worst assumption.

Reply:
If `Count` side-effects, would that not be a bug of the list implementation? 

I'm trying to figure out here if we're adding trivia into the code to optimize for an edge case and if so, how large is the edge case?

Reply:
> If Count side-effects, would that not be a bug of the list implementation?

Yes, but the JIT doesn't know that.

Either way, this is pretty minor.  I'd be totally fine with removing it.

Reply:
Understood. So in this specific code, that side effect would really be a (presumably nonexistent) bug in the implementation of `List<T>.Count`. This possible JIT issue would only exist if we changed to a different list implementation that had such a side effect bug.

I think we're safe here so I'd prefer to keep the code simpler, especially since we also will be supporting concurrency.